test/replication-luatest/gh_11574_hang_in_box_wait_quorum_test.lua

```diff
@@ -0,0 +1,128 @@
+local t = require("luatest")
+local server = require("luatest.server")
+local replicaset = require("luatest.replica_set")
+
+local g = t.group()
+
+g.before_all(function(g)
+    g.replicaset = replicaset:new({})
+    g.server_1_uri = server.build_listen_uri("server1", g.replicaset.id)
+    g.server_2_uri = server.build_listen_uri("server2", g.replicaset.id)
+    g.server_3_uri = server.build_listen_uri("server3", g.replicaset.id)
+
+    g.server1 = g.replicaset:build_and_add_server({
+        alias = "server1",
+        box_cfg = {
+            listen = g.server_1_uri,
+            replication = {g.server_1_uri, g.server_2_uri, g.server_3_uri},
+            replication_synchro_quorum = 4,
+            replication_timeout = 0.5,
+        },
+    })
+
+    g.server2 = g.replicaset:build_and_add_server({
+        alias = "server2",
+        box_cfg = {
+            listen = g.server_2_uri,
+            replication = {g.server_1_uri, g.server_2_uri, g.server_3_uri},
+            read_only = true,
+            election_mode = "manual",
+            replication_timeout = 0.5,
+        },
+    })
+
+    g.server3 = g.replicaset:build_and_add_server({
+        alias = "server3",
+        box_cfg = {
+            listen = g.server_3_uri,
+            replication = {g.server_1_uri, g.server_2_uri, g.server_3_uri},
+            read_only = true,
+            election_mode = "manual",
+            replication_timeout = 0.5,
+        },
+    })
+
+    g.replicaset:start()
+    g.replicaset:wait_for_fullmesh()
+end)
+
+g.after_all(function(g)
+    g.replicaset:drop()
+end)
+
+local function capture_synchro_queue_and_push_sync_transaction(server)
+    server:exec(function()
+        local fiber = require("fiber")
+        require('compat').replication_synchro_timeout = 'new'
+        if box.space.s == nil then
+            box.schema.create_space("s", {is_sync = true}):create_index("p")
```

Comment:
Let's move it to `before_all`, the space is used in all tests. The same applies to the compat